cpp/src/arrow/type_traits.h

```diff
@@ -644,6 +644,14 @@ template <typename T, typename R = void>
 using enable_if_physical_unsigned_integer =
     enable_if_t<is_physical_unsigned_integer_type<T>::value, R>;
 
+template <typename T>
+using is_physical_integer_type =
+    std::integral_constant<bool, is_physical_unsigned_integer_type<T>::value ||
+                                     is_physical_signed_integer_type<T>::value>;
```

Comment:
not sure 100%. but, is this indentation correct?

Reply:
C++ indentation is deferred to `clang-format`. We have a CI job which ensures that running `clang-format` is a no-op. As for explaining why it made this choice... I'm not familiar enough with its implementation to say, sorry